Automattic/jetpack
# Form Block: allowing marketing consent collection

**Changes proposed in this Pull Request:**
* Lets expand the form block, add a new option allowing folks to collect marketing consent for their contact forms.

**Jetpack product discussion**
N/A

**Does this pull request change what data or activity we track or use?**
* It adds 1 metadata field to the “feedback” post type.
*Testing instructions:*
## test case 1
1. Go to Posts > Add New
2. Search for ‘form’.
3. Select any of the form types
![image](https://user-images.githubusercontent.com/47211808/89404907-066b3700-d71b-11ea-98ec-a7f059463c56.png)

## test case 2
1. Perform test case 1.
2. Select the form block.
3. Click on add “add email permission request”.
![image](https://user-images.githubusercontent.com/47211808/89405039-361a3f00-d71b-11ea-9c37-a79b8975383c.png)
4. Select the “Mention that you can email”
![image](https://user-images.githubusercontent.com/47211808/89405072-47fbe200-d71b-11ea-8f3a-9e7c654f5404.png)
5. See the new implicit “consent” form element is added to your form.
![image](https://user-images.githubusercontent.com/47211808/89407322-dcb40f00-d71e-11ea-8040-23fcbd9fbc3f.png)
6. Go to the preview, see the new implicit “consent” form element is added to your form.
/Todo WIP/
7. Fill in a form entry, select the consent.
8. Find the new entry on the feedback page.
/Todo WIP/

## test case 3
1. Perform test case 1.
2. Select the form block.
3. Click on add “add email permission request”.
4. Select the “Add a privacy checkbox”
![image](https://user-images.githubusercontent.com/47211808/89407230-b4c4ab80-d71e-11ea-8d63-1f88cd63af91.png)
5. See the new explicit “consent” form element is added to your form.
![image](https://user-images.githubusercontent.com/47211808/89407348-eb022b00-d71e-11ea-81d5-f330054973c1.png)
6. Go to the preview, see the new implicit “consent” form element is added to your form.
/Todo WIP/
7. Fill in a form entry, select the consent.
8. Find the new entry on the feedback page.
/Todo WIP/
